src/main/java/seedu/organizer/model/task/Deadline.java

```diff
@@ -38,7 +46,32 @@ public Deadline(String deadline) {
      * Returns true if a given string is a valid task deadline.
      */
     public static boolean isValidDeadline(String test) {
-        return test.matches("") || test.matches(DEADLINE_VALIDATION_REGEX);
+        return test.matches(DEADLINE_VALIDATION_REGEX) && isValidDate(test);
+    }
+
+    /**
+     * Returns true if a given string is a valid date.
+     */
+    public static boolean isValidDate(String test) {
+        Logger logger = LogsCenter.getLogger(Deadline.class);
+
+        try {
+            String format = "yyyy-MM-dd";
+            SimpleDateFormat sdf = new SimpleDateFormat(format);
+            sdf.setLenient(false);
+            Date date = sdf.parse(test);
+
+            if (date.compareTo(new Date()) < 0) {
```

Comment:
Is this line really required? 
It seems that the method will return true even if `data.compareTo(new Date()) >= 0` as per line 65.